rita_client/src/exit_manager/mod.rs

```diff
@@ -590,11 +591,62 @@ fn initialize_selected_exit_list(exit: String, server: ExitServer) {
     });
 }
 
+/// Retrieve a unix timestamp from the exit's mesh IPv6
+pub async fn get_exit_time(exit: ExitServer) -> Option<i64> {
+    info!("Getting the exit time");
+    let exit_ip = exit.subnet.ip();
+    let exit_port = exit.registration_port;
+    let url = format!("http://[{}]:{}/time", exit_ip, exit_port);
+
+    let client = awc::Client::default();
+    let response = match client.get(&url).send().await {
+        Ok(mut response) => {
+            trace!("Response is {:?}", response.status());
+            trace!("Response is {:?}", response.headers());
+            response.json().await
+        }
+        Err(e) => {
+            error!("Failed to get exit time stamp {:?}", e);
+            return None;
+        }
+    };
+
+    let exit_time: ExitTimeStamp = match response {
+        Ok(a) => a,
+        Err(e) => {
+            error!("Failed to get exit time stamp {:?}", e);
+            return None;
+        }
+    };
+
+    Some(exit_time.secs_unix_epoch)
+}
+
+/// Set the router's time using "date -s \"@12345678\""
+pub async fn set_local_time(unix_time: i64) {
```

Comment:
this should be in a time file in althea kernel interface

Reply:
According to the docs on SystemTime, it looks like your function uses the standard way to get unix time from it, so I think it's a convenient utility fn.  I actually added a second fn for converting from a SystemTime to secs using the same approach.

Reply:
done